bin/parse_podfile_lock

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+
+require 'json'
+require 'yaml'
+
+dependency_metadata_regex = /(?<pod>.*?)\s\((=|(~>))?\s?(?<version>(\d*\.?)+)\)/
+
+podfile_lock = nil
```

Comment:
We can write this as follows to avoid assigning `nil` to podfile_lock

```
podfile_lock = begin
  YAML.safe_load(File.read(ARGV[0]))
rescue SyntaxError, Psych::SyntaxError
  warn 'Unable to parse Podfile.lock file'
  exit(1)
end
```

Following is a quick test.
```
irb(main):001:1* a = begin
irb(main):002:1*   4
irb(main):003:1* rescue StandardError => e
irb(main):004:0> end
=> 4
irb(main):005:0> a
=> 4
irb(main):006:1* a = begin
irb(main):007:1*   raise StandardError.new
irb(main):008:1* rescue StandardError => e
irb(main):009:0> end
=> nil
irb(main):010:0> a
=> nil
```

Reply:
There is a `to_yaml` method here
https://www.rubydoc.info/github/CocoaPods/Core/Pod/Lockfile#to_yaml-instance_method

Gerald, could you look into whether you should use `to_yaml` or `YAML.safe_load`?